Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -31,41 +31,64 @@ extension Style {
         }
     }
 
-    /// Filters through source to determine supported locale styles.
-    /// This is needed for v7 support
+    /// Returns the BCP 47 language tag supported by Mapbox Streets source v8 that is most preferred according to the given preferences.
+    /// Docs for language, region, and script codes: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html
+    internal func preferredMapboxStreetsLocalization(among preferences: [String]) -> String? {
+        let supportedLocaleIdentifiersv8 = ["ar", "en", "es", "fr", "de", "it", "pt", "ru", "zh-Hans", "zh-Hant", "ja", "ko", "vi"].map(Locale.init(identifier:))
+
+        let preferredLocales = preferences.map(Locale.init(identifier:))
+
+        let mostSpecificLanguage = Bundle.preferredLocalizations(from: supportedLocaleIdentifiersv8.map { $0.identifier },
+                                                                 forPreferences: preferences)
+            .max { $0.count > $1.count }
```

Comment:
I don't know if getting the most specific/prefered language by comparing the length is the good idea, would it make more sense if we get the first value of the returned `preferrredLocalizations(from:forPreferences)`, consider this will be an ordered list based on user's preferences?

> Return Value
> An array of locale identifiers, ordered according to user preference.
>
> If none of the user-preferred localizations are available, this method returns one of the values in localizationsArray.

Reply:
Updated in latest commit 👍

Reply:
The idea here was to prefer, say, `zh-Hant` over `zh`, all things considered. Be sure to test whether `preferrredLocalizations(from:forPreferences)` is actually ordered by user preference. I remember being surprised about something like that in the past.